fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -2126,6 +2129,61 @@ private void handleCtasRollback(TableName table) {
         }
     }
 
+    private void handleIotsStmt() {
+        InsertOverwriteTableSelect iotsStmt = (InsertOverwriteTableSelect) this.parsedStmt;
+        try {
+            // create table
+            DdlExecutor.execute(context.getEnv(), iotsStmt.getCreateTableLikeStmt());
+            context.getState().setOk();
+        } catch (Exception e) {
+            // Maybe our bug
+            LOG.warn("IOTS create table error, stmt={}", originStmt.originStmt, e);
+            context.getState().setError(ErrorCode.ERR_UNKNOWN_ERROR, "Unexpected exception: " + e.getMessage());
+        }
+        // after success create table insert data
+        if (MysqlStateType.OK.equals(context.getState().getStateType())) {
+            try {
+                parsedStmt = iotsStmt.getInsertStmt();
+                parsedStmt.setUserInfo(context.getCurrentUserIdentity());
+                execute();
+                if (MysqlStateType.ERR.equals(context.getState().getStateType())) {
+                    LOG.warn("IOTS insert data error, stmt={}", iotsStmt.toSql());
+                    handleIotsRollback(iotsStmt.getCreateTableLikeStmt().getDbTbl());
+                }
+            } catch (Exception e) {
+                LOG.warn("IOTS insert data error, stmt={}", iotsStmt.toSql(), e);
+                handleIotsRollback(iotsStmt.getCreateTableLikeStmt().getDbTbl());
+            }
+        }
+
+        // overwrite old table with new table
+        try {
+            AlterTableStmt alterTableStmt = iotsStmt.getAlterTableStmt();
+            DdlExecutor.execute(context.getEnv(), alterTableStmt);
+            context.getState().setOk();
```

Comment:
for user, he is executing an `insert` stmt, not a `create table` or `replace table` stmt.
so the result we return should exactly same as `insert`.
But here you call `context.getState().setOk()`, so the insert result is missing.